Packs/iDefense/Integrations/iDefense_v2/iDefense_v2.py

```diff
@@ -310,14 +318,190 @@ def uuid_command(client: Client, args: dict, reliability: DBotScoreReliability)
+                            analysis_info, doc_search_client, indicatorTypeHash=True)
+
+                        return Res, analysis_info, context
+
+
+def _enrich_analysis_result_with_intelligence(analysis_info, doc_search_client, indicatorTypeHash: bool = False):
+    """
+
+    Adds Intelligence reports and Intelligence alerts information to analysis result for the indicator using given doc search client                                        # noqa: E501
+
+    Args:
+        analysis_result obtained from _extract_analysis_info function call
+        client: ACTI Document search contoller client
+
+    Returns:
+        analysis_result enriched with intelligence alert and intelligence report information if available for the indicator
+
+    """
+
+    indicator = analysis_info['MD5'] if indicatorTypeHash else analysis_info['Name']
+    demisto.debug(f"getting ia for indicator {indicator}")
+
+    alerts, reports = _get_ia_for_indicator(indicator, doc_search_client)
+
+    if alerts is not None:
+        analysis_info['Intelligence Alerts'] = alerts if len(
+            alerts) > 0 else 'No Intelligence Alert has been linked to this indicator'
+    if reports is not None:
+        analysis_info['Intelligence Reports'] = reports if len(
+            reports) > 0 else 'No Intelligence Report has been linked to this indicator'
+
+    return analysis_info
+
+
+def _get_ia_for_indicator(indicator: str, doc_search_client: Client):
+    """
+    Perform document controller api call with given doc search client to get
+    Intelligence Alerts and Intelligence Reports for given indicator
+
+    Args:
+        client: ACTI Document search contoller client
+
+    Returns:
+        intelligence alert and intelligence report dictionaries if api has response else None
+
+    """
+
+    res = {}
+    intelligence_alerts, intelligence_reports = None, None
+
+    try:
+        res = doc_search_client.threat_indicator_search(
+            url_suffix='/v0', data={'type.values': ['intelligence_alert', 'intelligence_report'], 'links.display_text.query': indicator})                                                                       # noqa: E501
+
+        alerts = {item['title']: item['uuid'] for item in res.get('results', []) if item['type'] == 'intelligence_alert'}
+        reports = {item['title']: item['uuid'] for item in res.get('results', []) if item['type'] == 'intelligence_report'}
+        intelligence_alerts = {title: IDEFENSE_URL_TEMPLATE.format('intelligence_alert', uuid) for title, uuid in alerts.items()}
+        intelligence_reports = {title: IDEFENSE_URL_TEMPLATE.format(
+            'intelligence_report', uuid) for title, uuid in reports.items()}
+
+    except DemistoException as e:
+        demisto.debug(e.message)
+
+    except Exception as e:
+        demisto.debug(f"Unexpected error occured while calling document api for indicator {indicator}")
+        demisto.debug(e)
+
```

Comment:
please make sure every exception is raised (rather than handled gracefully) so the integration/playbook running it will fail, letting the users know of the error.

Reply:
So @dorschw do you want me to add another line in 496 like this : 
 _`raise DemistoException("")`_ 
this will help users know and they don't need us to look at the server logs

Reply:
or this

![image](https://user-images.githubusercontent.com/40510780/146357687-2997729e-d4b8-481f-b18f-5259db1f4929.png)

Reply:
I think it will be better to clean this part as we used this while we were developing and were seeing server logs

Reply:
This is for users who don't have access to our document API. To do that we log the errors and this helps us to create no errors and provide data without enrichment as they are not paying for our Document API

Reply:
https://xsoar.pan.dev/docs/integrations/code-conventions#exceptions-and-errors

Reply:
Hey Kobbi. We have introduced this for some different purpose to not add IA/IR to those customers who haven't bought Document API. Let me see if I can introduce that to show our customers this message.

Reply:
@satyakidroid not 100% sure I understood, but - what will happen if this function raises an unexpected error for users who bought the document API? our convention is to fail the command in this case, that's what the user would expect.

Reply:
Yep . I see . Will look into this issue 👍

Reply:
not failing completely as user will be using this enrichment only if the customer bought Document API access. I am thinking of adding some message the user will be able to see and understand.
This is when user has access to Document API :
![image](https://user-images.githubusercontent.com/40510780/146407472-133ddc04-5199-48de-9f91-f17c260bbb1a.png)

This is when user don't have access to Document API :
![image](https://user-images.githubusercontent.com/40510780/146407657-403770f1-d048-482e-bd5d-6f0c63b0f962.png)

And generally i agree that if some other error occurs we will be catching that too and will be hiding that which is not at all good